go/merge/three_way_list.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package merge
+
+import (
+	"fmt"
+
+	"github.com/attic-labs/noms/go/types"
+)
+
+func threeWayListMerge(a, b, parent types.List) (merged types.List, err error) {
+	aSpliceChan, bSpliceChan := make(chan types.Splice), make(chan types.Splice)
+	aStopChan, bStopChan := make(chan struct{}, 1), make(chan struct{}, 1)
+
+	go func() {
+		a.Diff(parent, aSpliceChan, aStopChan)
+		close(aSpliceChan)
+	}()
+	go func() {
+		b.Diff(parent, bSpliceChan, bStopChan)
+		close(bSpliceChan)
+	}()
+
+	stopAndDrain := func(stop chan<- struct{}, drain <-chan types.Splice) {
+		close(stop)
+		for range drain {
+		}
+	}
+
+	defer stopAndDrain(aStopChan, aSpliceChan)
+	defer stopAndDrain(bStopChan, bSpliceChan)
+
+	zeroSplice := types.Splice{}
+	zeroToEmpty := func(sp types.Splice) types.Splice {
```

Comment:
It would be nice to have a quick comment here, because I needed to look up the splice code to understand this. Namely, that the zero actually means something to the splice code, so it uses "unassigned" instead.